_posts/know-your-bounds.md

```diff
@@ -0,0 +1,126 @@
+
+Everyone can't be wrong, right?
+
+The same goes with big data, microservices, or containers, to name a few. If everybody says they are good, they must be, and moreover, they must be the solution to all my problems.
+
+That is how it starts. Instead of looking at the business problem, we let the technical hype or cool new kid on the block take over.
+
+### Intangibles come for free
+
+I also believe that human beings can estimate the value of tangibles but not so quickly intangibles. For example, when presented with the cost of a building, we can see the dollar value. When presented with the "cost" of time, we struggle to make the connection. It's trivially easy to observe people investing a tremendous amount of time addressing issues with a low or negative return on the investment (ROI).
+
+> a few years ago, an acquaintance drove 300 kilometers to save a few bucks on a new phone.
+
+I observed this effect in two different settings in the last twenty years. 
+
+Due to our inability to estimate intangibles, we tend to underestimate the cost of changing when there is no perception of physical change. That manifests itself with the last famous words, "how long could it take" or "how hard can it be." A typical example in the software engineering world is the decision to upgrade a dependency to its latest release, even if we're aware of the many breaking changes and there is no compelling reason. I witnessed teams moving from Entity Framework to Entity Framework Core, assuming it was still Entity Framework, even if the project didn't benefit from the upgrade.
+
+The presented behavior, to some extent, is supported by customers or users. How often have we heard statements like "for you, it's certainly easy"? The assumption here is that software is a toy we play with, and everything is easy. It's another form of the "intangibles come for free" mindset.
+
+### We can design for changes
+
+We have a crystal ball. We know what the system scalability requirements will look like in three years, and we'll shape today's architecture to accommodate the future. Sure, you bet!
+
+How often have you prematurely designed an extension point that later turned out to be completely useless? I did it many times. Those are instances of over-engineering the design, assuming we can accommodate future changes even if we have no idea what they'll be.
+
+That's where the myth of infinite scalability is rooted.
+
+## About knowing your bounds
+
+You just got out of college, and you're looking for your first car. Would you buy a seven-seat car because maybe you'll have a seven-people family and a dog in ten years or would you go with a small used one because you have no idea where you will be next month?
```

Comment:
```suggestion
You just got out of college, and you're looking for your first car. Would you buy a seven-seat car because maybe you'll have a seven-person family and a dog in ten years or would you go with a small used one because you have no idea where you will be next month?
```